src/http.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::{self, Write};
```

Comment:
here you start a hyper server, so how user embed this lib? if  user has already used hyper or iron outside and want to register our handler in `/metrics` or other path, how do user do?